pkg/reconciler/certificate/resources/cert_manager_certificate.go

```diff
@@ -46,94 +43,72 @@ const (
 func MakeCertManagerCertificate(cmConfig *config.CertManagerConfig, knCert *v1alpha1.Certificate) (*cmv1.Certificate, *apis.Condition) {
 	var commonName string
 	var dnsNames []string
-	attemptedToShorten := false
 
 	if len(knCert.Spec.DNSNames) > 0 {
 		commonName = knCert.Spec.DNSNames[0]
 	}
 
-	// https://github.com/knative-sandbox/net-certmanager/issues/214
-	// Only use the domain template if the commonName is too big.
+	// Only attempt to do something special if the entry from DNSNames[0] is too big.
 	// This is to make the upgrade path easier and reduce churn on certificates.
 	// The Route controller adds spec.domain to existing KCerts
 	// The KCert controller requests new certs with same domain names, but a different CN if spec.domain is set and the other domain name would be too long
 	// cert-manager Certificates are updated only if the existing domain name kept them from being issued.
 	if len(commonName) > longest {
-		if knCert.Spec.Domain != "" && knCert.Spec.Domain != commonName {
-			//Split out the domain, and create a hash of the remaining part
-			domainSuffix := "." + knCert.Spec.Domain
-			prefix := strings.TrimSuffix(commonName, domainSuffix)
-			if len(prefix) > base36Len {
-				attemptedToShorten = true
-
-				parsedUUID, err := uuid.Parse(string(knCert.UID))
-				if err != nil {
+		//if we have a domain field, we can attempt to shorten, or check if we are dealing with a domainMapping
+		if knCert.Spec.Domain != "" {
+			// if the domain and commonName pulled from DNSNames are the same, we are dealing with a domainmapping
+			if knCert.Spec.Domain == commonName {
+				if len(commonName) > longest {
```

Comment:
We can technically drop this conditional since it's the same as line 56 - so it's always true